File: Beakjoon/14467.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException{
        int[] arr = new int[10];
        Arrays.fill(arr, -1);
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(bf.readLine().trim());
        int num1, num2, answer = 0;
        for(int i=0;i<N;i++){
            StringTokenizer st = new StringTokenizer(bf.readLine().trim());
            num1 = Integer.parseInt(st.nextToken()) - 1;
            num2 = Integer.parseInt(st.nextToken());
            if(arr[num1] != -1 && arr[num1] != num2){
                answer++;
            }
            arr[num1] = num2;
        }
        System.out.println(answer);
    }
}

File: Beakjoon/14467.cpp
#include <vector>
#include <iostream>

using namespace std;

int main(){
    int N, num1, num2, answer = 0;  
    cin>>N;

    vector<vector<int>> hist;
    hist.resize(10); 
    for(int i=0;i<N;i++){
        cin>>num1>>num2;

        if(!hist[num1-1].empty() && hist[num1-1].back() != num2){
            answer++;
        }
        hist[num1-1].push_back(num2);
    }
    cout<<answer;
}